Shared/Loader.swift

```diff
@@ -26,8 +26,12 @@ open class Loader<T, ListenerType: LoaderListener> where T == ListenerType.T {
     }
 
     open func load(_ data: T) {
-        for listener in listeners {
-            listener.loader(dataLoaded: data)
+        if !Thread.isMainThread {
```

Comment:
Hold on, I'm going to make you cry.

http://blog.benjamin-encz.de/post/main-queue-vs-main-thread/

Reply:
Ha! Yeah, I read about `sync` running on main thread unexpectedly when you mentioned it before. I would only ever use `sync` in extreme cases (typically low level), like how we use it in DB layer. Anyhow, there is no reason here not to '.main.async {' in all cases and remove the if. There is almost never a rationale for `if isMain { do this } else { do this }` in front-end code except if one needed to band-aid some odd case.